### PR TITLE
Automated Workflow Executions Model

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Adds in `AutomatedWorkflowExecution` model which will be used to setup automated pipelines for when Samples have pair end data added to them and a Project has an automated pipeline setup.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

Run the tests and verify they pass

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/models/automated_workflow_execution.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  def validate_namespace_type
+    # Only Projects should have automated workflow executions
+    return if %w[Project].include?(namespace.type)
+
+    errors.add(namespace.type, 'namespace cannot have automated workflow executions')
+  end
+
+  def higher_access_level_than_group
+    return unless highest_group_member && highest_group_member.access_level > access_level
```
Comment (deepsidhu85) [inline app/models/automated_workflow_execution.rb:28]: Was this method accidentally copied here? I don't see it being used for anything

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks good just the one comment below

File: app/models/automated_workflow_execution.rb (hunk 1 above)
Comment (ericenns) [reply app/models/automated_workflow_execution.rb]: Good catch, removed in https://github.com/phac-nml/irida-next/pull/568/commits/b4b6e5c13d1bbac6a5a1209ba437d68d707f9405

Comment (deepsidhu85) [reply app/models/automated_workflow_execution.rb]: Thanks!

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Pull request merged**